src/Symfony/Component/Dotenv/Dotenv.php

```diff
@@ -47,33 +47,38 @@
      */
     public function load(string $path, string ...$paths): void
     {
-        array_unshift($paths, $path);
-
-        foreach ($paths as $path) {
-            if (!is_readable($path) || is_dir($path)) {
-                throw new PathException($path);
-            }
+        $this->doLoad(false, $path, ...$paths);
+    }
 
-            $this->populate($this->parse(file_get_contents($path), $path));
-        }
+    /**
+     * Loads one or several .env files and enables override existing vars.
+     *
+     * @param string    $path  A file to load
+     * @param ...string $paths A list of additional files to load
+     *
+     * @throws FormatException when a file has a syntax error
+     * @throws PathException   when a file does not exist or is not readable
+     */
+    public function overload(string $path, string ...$paths): void
+    {
+        $this->doLoad(true, $path, ...$paths);
     }
 
     /**
      * Sets values as environment variables (via putenv, $_ENV, and $_SERVER).
      *
-     * Note that existing environment variables are not overridden.
-     *
-     * @param array $values An array of env variables
+     * @param array $values               An array of env variables
+     * @param bool  $overrideExistingVars true when existing environment variables must be overridden
      */
-    public function populate(array $values): void
+    public function populate(array $values, bool $overrideExistingVars = false): void
```

Comment:
This should be reverted IMHO: it makes no sense to provide two public interfaces to do the same thing.

Reply:
I agree but I think the same thing of `populate()` method, I don't know why it is public because the purpose of `Dotenv` is to use a file, no directly an array of vars it's so low level IMHO. So I added here to be consistent but let me know if you have no doubt about this revert, I can do it :)